### DISCUSSION TITLE
Logging disappeared when calling DataFrame API in tests

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#656] [Q&A]
I tried to add some logs in the tests that parsing the csv file. There is an interesting issue I found. My project tech stack is a Spring Boot/R2dbc/Kotlin Coroutinnes, by default I used sfl4j API and Logback in the background.

In the unit tests for parsing csv using DataFrame, the `log.debug` does not print the logging stack at all, but `print` works well.

```kotlin 
class MyDataReaderTest{
    companion object {
        private val log = LoggerFactory.getLogger(MyDataReaderTest::class.java)
    }

    private val processor = MyDataReader()

    @Test
    fun `parsing sample csv file`() {
        val fileName = "..."
        val result = processor.read(ClassPathResource(fileName).file.readBytes())
        // log.debug("parsed result: $result")
        println("parsed result: $result")
        result.timesheetItems.size shouldBe 4
    }
}
```

In other tests, such my Controller, Repository etc, that does not contain the DataFrame part, the logging is working well.

I am not sure what is root cause of this issue.

**Comment by koperagen** [comment]
Hi! Can it be that logger is not configured for specific test class? Or, to put it another way, if you add this test to one of those other tests you have, will logging stop working there too?

**Reply by hantsy** [reply]
@koperagen I updated the original post, and add an dummy test example(which is modified from my real example, just removed the business related words)
